sycl/test-e2e/Graph/Explicit/buffer_copy_host2target_offset.cpp

```diff
@@ -0,0 +1,54 @@
+// REQUIRES: level_zero, gpu
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+// RUN: %if ext_oneapi_level_zero %{env ZE_DEBUG=4 %{run} %t.out 2>&1 | FileCheck %s %}
+//
+// CHECK-NOT: LEAK
+
+// Tests adding buffer copy offset
+// -- Host to Target (write path) --  nodes
+// using the explicit API and submitting
+// the graph.
+// The second run is to check that there are no leaks reported with the embedded
+// ZE_DEBUG=4 testing capability.
+
+#include "../graph_common.hpp"
+
+int main() {
+  queue Queue;
+
+  using T = int;
+
+  int Offset = 100;
+  T DataA[Size + Offset], DataB[Size];
+  std::iota(DataA, DataA + Size + Offset, 1);
+  std::iota(DataB, DataB + Size, 1000);
+
+  T ReferenceA[Size + Offset];
+  for (size_t i = 0; i < Size + Offset; i++) {
+    if (i < Offset)
+      ReferenceA[i] = DataA[i];
+    else
+      ReferenceA[i] = DataB[i - Offset];
+  }
+
+  exp_ext::command_graph Graph{Queue.get_context(), Queue.get_device()};
+
+  buffer<T, 1> BufferA(DataA, range<1>(Size + Offset));
+  BufferA.set_write_back(false);
+
+  auto NodeA = Graph.add([&](handler &Cgh) {
```

Comment:
```suggestion
  auto NodeA = Graph.add([&](handler &CGH) {
```
Should use consistent naming, also applies to many of the other new tests.